src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs

```diff
@@ -194,9 +263,35 @@ private void WriteRawValueCore(ReadOnlySpan<byte> utf8Json, bool skipInputValida
             }
 
             utf8Json.CopyTo(output.Slice(BytesPending));
-            BytesPending += len;
+            BytesPending += (int)len;
 
             SetFlagToAddListSeparatorBeforeNextItem();
         }
+
+        private void GrowIfNeeded(int len)
+        {
+            int maxRequired = len + 1; // Optionally, 1 list separator. We've guarded against integer overflow earlier in the call stack.
+
+            if (_memory.Length - BytesPending < maxRequired)
+            {
+                Grow(maxRequired);
+            }
+        }
+
+        private static JsonTokenType ValidateJson(in ReadOnlySpan<byte> utf8Json)
```

Comment:
The `ValidateJson` methods don't seem particularly justified given that they're only being called once each.

Reply:
Removed.